### PR TITLE
Ready for Review bingo box

### PR DESCRIPTION
**Pull request by hkf-tech**
Ready for Review bingo box changes. 
New page created on the Employer V2. V1 will now redirect to V2 when ready for review is clicked provided that they have enhanced employer permission.

### CODE REVIEW

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Cohort/ReviewViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Models.Cohort
+{
+    public class ReviewViewModel
+    {
+        public string AccountHashedId { get; set; }
+
+        public string BackLinkUrl { get; set; }
+
+        public IEnumerable<ReviewCohortSummaryViewModel> CohortSummary { get; set; }
```
Comment (chrisfoster76) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Cohort/ReviewViewModel.cs]: This seems a strange name for this property. Maybe "Cohorts" would be fine?

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+﻿@using SFA.DAS.Encoding
+@model SFA.DAS.EmployerCommitmentsV2.Web.Models.Cohort.ReviewViewModel
+@inject IEncodingService EncodingService
```
Comment (chrisfoster76) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: As on Draft page, we could encode in the view model mapper

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    @foreach (var cohort in Model.CohortSummary)
+                    {
+
+                    <tr class="govuk-table__row">
+                        <td class="govuk-table__cell" data-label="Training provider">@cohort.ProviderName</td>
+                        <td class="govuk-table__cell" data-label="Reference">@cohort.CohortReference</td>
+                        <td class="govuk-table__cell govuk-table__cell--numeric" data-label="Number of apprentices">@cohort.NumberOfApprentices</td>
+                        <td class="govuk-table__cell" data-label="Last Message">@cohort.LastMessage</td>
+                        <td class="govuk-table__cell govuk-table__cell--numeric" data-label="">
+                            @Html.ActionLink("Details", "Details", "Cohort", new { @Model.AccountHashedId, cohort.CohortReference }, new { @class = "govuk-link", id = "return-link" })
```
Comment (chrisfoster76) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: As on Draft page, having a repeating Id should be avoided and the proposed Id is wrong. We could just add a class to help automation testing.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ReviewRequestViewModelMapper.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        public async Task<ReviewViewModel> Map(ReviewRequest source)
+        {
+            var cohortsResponse = await _commitmentsApiClient.GetCohorts(new GetCohortsRequest { AccountId = source.AccountId });
+
+            var reviewViewModel =  new ReviewViewModel
+            {
+                AccountHashedId = source.AccountHashedId,
+                BackLinkUrl = _linkGenerator.Cohorts(source.AccountHashedId),
+                CohortSummary = cohortsResponse.Cohorts
+                .Where(x => x.WithParty == Party.Employer && !x.IsDraft)
```
Comment (chrisfoster76) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ReviewRequestViewModelMapper.cs]: As with Draft, we could use an extension method to determine cohort "status"

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Cohort/ReviewViewModel.cs (hunk 1 above)
Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Cohort/ReviewViewModel.cs]: Done.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml (hunk 2 above)
Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: Oops, I moved the code to mapper, but forgot to remove the injection, nice spot thanks.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml (hunk 3 above)
Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: Agreed, the Id shouldn't repeat, ( copy&paste gone wrong :). As suggested I created a class.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ReviewRequestViewModelMapper.cs (hunk 4 above)
Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ReviewRequestViewModelMapper.cs]: Created an extension function.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+﻿@using SFA.DAS.Encoding
```
Comment (chrisfoster76) [inline src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: This is no longer used.

Comment (hkf-tech) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Review.cshtml]: Yes. I will remove it. Thanks

**Pull request merged**